docs/source/reference/ha.rst

```diff
@@ -0,0 +1,301 @@
+:orphan:
+
+High availability deployment
+============================
+
+|st2| has been systematically built with High availability(HA) as a goal. The exact deployment
+steps to achieve HA depend on the specifics of the infrastructure in which |st2| is deployed. This
+guide covers a brief explanation on various |st2| services, how they interact and the external
+services necessary for |st2| to function.
+
+In the section :doc:`/install/overview` a detailed picture and explanation on how single-box deployments
+work is provided. Repeating the picture here to keep context and use as a reference to layer on some
+HA deployment specific details.
+
+.. figure :: /_static/images/st2-deployment-big-picture.png
+    :align: center
+
+.. source https://docs.google.com/drawings/d/1X6u8BB9bnWkW8C81ERBvjIKRfo9mDos4XEKeDv6YiF0/edit
+
+Components
+----------
+
+st2api
+^^^^^^
+This process host the REST API endpoints that serve requests from  WebUI, CLI and ChatOps. It maintains
+connections to MongoDB to store and retrieve information. It also connects to RabbitMQ to push various
+kind of messages onto the message bus. It is a gunicorn managed process which by default listens on
+port ``9101`` and is front-ended by Nginx acting as a reverse proxy.
+
+n st2api processes can be behind a loadbalancer in an active-active configuration. Each of these
+processes can be deployed on separate compute instances.
+
+st2auth
+^^^^^^^
+All authentication is managed by this process. This process needs connection to MongoDB and an authentication
+backend. See `Authentication backends <ref-auth-backends>` for more information. It is a gunicorn
+managed process which by default listens on port ``9102`` and is front-ended by Nginx acting as a reverse proxy.
+
+n st2auth processes can be behind a loadbalancer in an active-active configuration. Each of these processes
+can be deployed on separate compute instances. If using the PAM authentication backend special care has to be
+taken to guarantee that all boxes on which an instance of st2auth runs should have the same users. Generally,
+all st2auth process should see the same identities, via some provider if applicable, for the system to work
+predictably.
+
+st2stream
+^^^^^^^^^
+This process exposes a server-sent event stream. Requires access to both MongoDB and RabbitMQ. It is also a
+gunicorn managed process, listens on port ``9102`` by default and is front-ended by Nginx acting as a reverse
+proxy. Clients like WebUI and ChatOps maintain a persistent connection with an st2stream process and receive
```

Comment:
Yeah, maybe a good idea to also clarify on that and contrast it to st2api (short lived vs long lived / opened connections and requests).

That's something which needs to be taken into account when doing a custom deployments since you need to configure appropriate timeouts for lb, wsgi app server, etc.